src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -2,14 +2,8 @@
   (:require [puppetlabs.trapperkeeper.core :as tk]
             [puppetlabs.services.protocols.request-handler :as handler]
             [puppetlabs.services.request-handler.request-handler-core :as core]
-            [puppetlabs.services.jruby.jruby-puppet-service :as jruby]
             [puppetlabs.trapperkeeper.services :as tk-services]))
 
-(defn- handle-request
-  [request jruby-service config]
-  (jruby/with-jruby-puppet jruby-puppet jruby-service
-    (core/handle-request request jruby-puppet config)))
-
```

Comment:
Yeah, in this case we either err on the side of putting a "service" thing in core code, or do code things in service code. This solution ended up being the least ugly, in my opinion.